testsuite/MDAnalysisTests/guesser/test_base.py

```diff
@@ -101,6 +104,131 @@ def test_partial_guess_attr_with_unknown_no_value_label(self):
         u = mda.Universe(top, to_guess=['types'])
         assert_equal(u.atoms.types, ['', '', '', ''])
 
+    def test_guess_topology_objects_existing_read(self):
+        u = mda.Universe(datafiles.CONECT)
+        assert len(u.atoms.bonds) == 72
+        assert list(u.bonds[0].indices) == [623, 630]
+
+        # delete some bonds
+        u.delete_bonds(u.atoms.bonds[:10])
+        assert len(u.atoms.bonds) == 62
+        # first bond has changed
+        assert list(u.bonds[0].indices) == [1545, 1552]
+
+        all_indices = [tuple(x.indices) for x in u.bonds]
+        assert (623, 630) not in all_indices
+
+        # test guessing new bonds doesn't remove old ones
```

Comment:
Could you also add a check that you don't duplicate existing bonds? `_add_bonds` should avoid this, but it's worth double checking.

Reply:
Done, although new test checks that the number of the bonds just doesn't change (since they start out with two of them in the first place)

Reply:
Added a part that tests whether the bond is guessed or not. I check the topology manually (`u._topology.bonds._guessed`) but `u.atoms[[1545, 1552]].bond.is_guessed` for some reason does think it's guessed, so something is getting lost in translation there. Not sure if that's a new issue or something here.

Reply:
If I'm following this properly, this seems like an issue?

Reply:
Yeah maybe. If it is I would guess it's pre-existing. Either way I'm not sure how much time I have for this and when, will try to get to it by the end of the week.

Reply:
Never mind, I just temporarily forgot how bonds worked. Fixed the test.